java/semantickernel-api/src/main/java/com/microsoft/semantickernel/Kernel.java

```diff
@@ -65,10 +83,20 @@ public Kernel(
      *
      * @return The fluent builder for creating a new instance of {@code Kernel}.
      */
-    public static Kernel.Builder builder() {
+    public static Builder builder() {
         return new Kernel.Builder();
     }
 
+    /**
+     * Creates a Builder that can create a copy of the current instance of {@code Kernel}. Use this
+     * method if you wish to modify the state of the kernel such as adding new plugins or services.
+     *
+     * @return A Builder that can create a copy of the current instance of {@code Kernel}.
+     */
+    public Builder copy() {
```

Comment:
I'd expect 'copy' to return a Kernel, not a Kernel.Builder. Maybe 'toBuilder' or a static 'Builder from(Kernel)' method in Kernel.Builder itself instead. I have a feeling this will be a pattern we repeat. Maybe a 'toBuilder' API in Buildable is in order?